### PR TITLE
Make scanner processor generic to scan all kind of uploads

### PR DESCRIPTION
**Pull request by ThomasKranitsas**

### CODE REVIEW

File: src/services/ProcessorService.js [hunk 1]
```diff
@@ -48,10 +90,12 @@ processScan.schema = {
...
       payload: Joi.object()
         .keys({
-          submissionId: Joi.string().required(),
           url: Joi.string().required(),
+          cleanDestinationBucket: Joi.string().default(config.get('aws.DEFAULT_CLEAN_BUCKET')).valid(config.get('WHITELISTED_CLEAN_BUCKETS')),
+          quarantineDestinationBucket: Joi.string().default(config.get('aws.DEFAULT_QUARANTINE_BUCKET')),
           fileName: Joi.string().required(),
-          status: Joi.string().required(),
+          callbackUrl: Joi.string(),
+          callbackTopic: Joi.string().default(config.get('AVSCAN_TOPIC')),
```
Comment (eisbilir) [inline src/services/ProcessorService.js:98]: we should also whitelist the topics

File: src/services/ProcessorService.js [hunk 2]
```diff
@@ -1,42 +1,84 @@
...
+    }
+  }
+  // Notify the caller
+  if (message.payload.callbackTopic) {
+    message.originator = config.get('KAFKA_ORIGINATOR');
+    message.topic = message.payload.callbackTopic;
+    await helper.postToBusAPI(message);
+  }
+  if (message.payload.callbackUrl) {
+    await helper.postToCallbackURL(_.omit(message.payload, ['callbackUrl', 'callbackTopic', 'cleanDestinationBucket', 'quarantineDestinationBucket']));
```
Comment (eisbilir) [inline src/services/ProcessorService.js:53]: this seems not implemented

**Review by eisbilir — CHANGES_REQUESTED** [review]

File: src/services/ProcessorService.js (hunk 1 above)
Comment (ThomasKranitsas) [reply src/services/ProcessorService.js:98]: Not really, the result can be posted anywhere depending on how the caller needs to use it

File: src/services/ProcessorService.js (hunk 2 above)
Comment (ThomasKranitsas) [reply src/services/ProcessorService.js:53]: Oh, shoot! I forgot to commit that 🤦‍♂️ ... lemme commit that

File: src/services/ProcessorService.js [hunk 3]
```diff
@@ -1,42 +1,84 @@
...
 const logger = require("../common/logger");
 const helper = require("../common/helper");
+const config = require("config");
+
+async function handleResult(message) {
+  // Move the file to the appropriate bucket
+  if (message.payload.isInfected) {
+    if (message.payload.quarantineDestinationBucket) {
+      // make sure the quarantine bucket is not in the whitelist
+      if (config.get('WHITELISTED_CLEAN_BUCKETS').includes(message.payload.quarantineDestinationBucket)) {
```
[the remark this answers is not in the document]
Comment (ThomasKranitsas) [reply src/services/ProcessorService.js:15]: It's defaulted but not validated. Only the clean destination bucket is validated

File: src/services/ProcessorService.js [hunk 4]
```diff
@@ -1,42 +1,84 @@
...
+        config.get('aws.DMZ_BUCKET'),
+        message.payload.fileName,
+        message.payload.quarantineDestinationBucket,
+        message.payload.fileName
+      );
+      // Update the URL
+      message.payload.url = `https://s3.amazonaws.com/${message.payload.quarantineDestinationBucket}/${message.payload.fileName}`;
+    } else {
+      // throw error about missing quarantine bucket
+      throw new Error(`File ${message.payload.fileName} is infected but no quarantine bucket is configured`);
```
[the remark this answers is not in the document]
Comment (eisbilir) [reply src/services/ProcessorService.js:29]: what if the requester doesn't want the file to be moved by this service?

File: src/common/helper.js [hunk 5]
```diff
@@ -167,11 +167,31 @@ async function moveFile (sourceBucket, sourceKey, targetBucket, targetKey) {
...
+ * @param{Object} message Body of the request to be Posted
+ * @returns {Promise}
+ */
+async function postToCallbackURL (message) {
+  const token = await getM2Mtoken();
+  const reqBody = {
+    method: "POST",
+    url: message.callbackUrl,
+    headers: {
+      Authorization: `Bearer ${token}`,
```
Comment (eisbilir) [inline src/common/helper.js:181]: there is no guarantee that auth method will work.

**Pull request merged**